src/key.rs

```diff
@@ -670,6 +672,21 @@ impl Connection {
             _ => (),
         }
 
+        #[cfg(target_os = "linux")]
+        let maybe_ring = KeyRing::from_special_id(KeyRingIdentifier::Session, false);
+        #[cfg(target_os = "linux")]
+        let description = format!("age-plugin-yubikey-{}-{}", self.yubikey.serial().0, self.slot);
+        #[cfg(target_os = "linux")]
+        if let Ok(ring) = maybe_ring {
+            if let Ok(key) = ring.search(&description) {
+                if let Ok(cached_pin) = key.read_to_vec() {
+                    if self.yubikey.verify_pin(&cached_pin).is_ok() {
```

Comment:
Any PIN caching _MUST_ respect the configured policy for the slot. If the slot uses `PinPolicy::Always`, then we must request the PIN directly from the user, and never cache it. What that means in practice is that the only time this cache should be used is `PinPolicy::Once`.